pkg/cluster/acrtoken.go

```diff
@@ -56,3 +61,51 @@ func (m *manager) ensureACRToken(ctx context.Context) error {
 
 	return nil
 }
+
+func (m *manager) updateACRToken(ctx context.Context) error {
+	// we do not want to rotate tokens in local development
+	// TODO: check for using rh-dev-env-pull on arointsvc along with or instead of local development
+	if m.env.IsLocalDevelopmentMode() {
+		return nil
+	}
+
+	tokenManager, err := acrtoken.NewManager(m.env, m.localFpAuthorizer)
+	if err != nil {
+		return err
+	}
+
+	rp := tokenManager.GetRegistryProfile(m.doc.OpenShiftCluster)
+	if rp == nil {
+		// shouldn't get here but return error if we do
+		return nil // TODO: add some error
+	}
+
+	err = tokenManager.RotateTokenPassword(ctx, rp)
+	if err != nil {
+		return err
+	}
+	// update cluster pull secret in openshift-azure-operator namespace
+	// secret is stored as a .dockerconfigjson string in the .dockerconfigjson key
+	encodedDockerConfigJson, _, err := pullsecret.SetRegistryProfiles("", rp)
+	if err != nil {
+		return err
+	}
+
+	// wait for response from operator that reconciliation is completed successfully
```

Comment:
The pull secret stored in the `cluster` secret is the source of truth for the cluster's pull credentials. If there is a difference between what is in `pull-secret` and what is in `cluster`, then whatever is in `cluster` wins.

Further, the operator merges the ARO pull secret with whatever custom pull secrets may have been provided by the customer for their own registries during its reconciliation to build the final cluster pull secret. If we update it ourselves we would have to re-implement that logic, so having the operator handle it makes the most sense.

Reply:
> a situation where operator is unable to run due to bad ACR token, and RP can't fix it because the operator isn't running.

To be honest I'm not sure under what circumstances the operator would stop working. I can do some investigation into it and get back to you.

> Also, curious on if we have a way we can breakglass rotate if we end up in a similar situation.

Breakglass shouldn't be difficult conceptually, it would just be manually creating a new password through the Azure portal or the CLI and updating the cluster secret. I can get some steps written up on the ADO wiki.